protocols/autonat/src/v2/client/behaviour.rs

```diff
@@ -148,19 +151,27 @@ where
         event: <Self::ConnectionHandler as ConnectionHandler>::ToBehaviour,
     ) {
         let (nonce, outcome) = match event {
-            Either::Right(nonce) => {
+            Either::Right(IncomingNonce { nonce, sender }) => {
                 let Some((_, info)) = self
                     .address_candidates
                     .iter_mut()
                     .find(|(_, info)| info.is_pending_with_nonce(nonce))
                 else {
                     tracing::warn!(%peer_id, %nonce, "Received unexpected nonce");
```

Comment:
We can remove this now and instead print the error that we send through the channel.

Reply:
I hope I changed it to your satisfaction.